jetty-websocket/websocket-jetty-client/src/main/java/org/eclipse/jetty/websocket/client/WebSocketClient.java

```diff
@@ -64,61 +63,27 @@ public class WebSocketClient extends ContainerLifeCycle implements WebSocketPoli
     private final List<WebSocketSessionListener> sessionListeners = new CopyOnWriteArrayList<>();
     private final SessionTracker sessionTracker = new SessionTracker();
     private final Configuration.ConfigurationCustomizer configurationCustomizer = new Configuration.ConfigurationCustomizer();
-    private final WebSocketComponents components;
     private boolean stopAtShutdown = false;
     private long _stopTimeout = Long.MAX_VALUE;
 
     /**
-     * Instantiate a WebSocketClient with defaults
+     * Instantiates a WebSocketClient with a default {@link HttpClient}.
      */
     public WebSocketClient()
     {
         this(null);
     }
 
     /**
-     * <p>
-     * Instantiate a WebSocketClient.
-     * </p>
+     * <p>Instantiates a WebSocketClient with the given {@link HttpClient}.</p>
      *
-     * <p>
-     *     HTTP behaviors of the WebSocket upgrade are taken from the HttpClient configuration.
-     * </p>
-     *
-     * @param httpClient the HttpClient to base internal defaults off of
+     * @param httpClient the HttpClient to use
      */
     public WebSocketClient(HttpClient httpClient)
     {
-        this (httpClient,
-            new WebSocketComponents(
-                null,
-                null,
-                httpClient != null ? httpClient.getByteBufferPool() : null,
-                null,
-                null,
-                httpClient != null ? httpClient.getExecutor() : null)
-        );
-    }
-
-    /**
-     * <p>
-     * Instantiate a WebSocketClient.
-     * </p>
-     *
-     * <p>
-     *     HTTP behaviors of the WebSocket upgrade are taken from the {@link HttpClient} configuration.
-     *     WebSocket behaviors are taken from the {@link WebSocketComponents} configuration.
-     * </p>
-     *
-     * @param httpClient the HttpClient to use for the HTTP behaviors of WebSocket upgrade
-     * @param webSocketComponents the WebSocketComponents to use for WebSocket behaviors
-     */
-    public WebSocketClient(HttpClient httpClient, WebSocketComponents webSocketComponents)
```

Comment:
Why did you remove this constructor?
It was added specifically to support this use case.

Reply:
@joakime read the comment in the issue. The addition of that constructor was wrong.